src/main/java/at/hannibal2/skyhanni/features/misc/customscoreboard/ScoreboardElements.kt

```diff
@@ -0,0 +1,633 @@
+package at.hannibal2.skyhanni.features.misc.customscoreboard
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.config.features.gui.customscoreboard.DisplayConfig.ArrowAmountDisplay
+import at.hannibal2.skyhanni.data.BitsAPI
+import at.hannibal2.skyhanni.data.HypixelData
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.MaxwellAPI
+import at.hannibal2.skyhanni.data.MayorAPI
+import at.hannibal2.skyhanni.data.PartyAPI
+import at.hannibal2.skyhanni.data.PurseAPI
+import at.hannibal2.skyhanni.data.QuiverAPI
+import at.hannibal2.skyhanni.data.QuiverAPI.asArrowPercentage
+import at.hannibal2.skyhanni.data.QuiverAPI.getArrowByNameOrNull
+import at.hannibal2.skyhanni.data.ScoreboardData
+import at.hannibal2.skyhanni.data.SlayerAPI
+import at.hannibal2.skyhanni.features.misc.customscoreboard.CustomScoreboardUtils.formatNum
+import at.hannibal2.skyhanni.features.misc.customscoreboard.CustomScoreboardUtils.getGroupFromPattern
+import at.hannibal2.skyhanni.features.misc.customscoreboard.CustomScoreboardUtils.getTitleAndFooterAlignment
+import at.hannibal2.skyhanni.mixins.hooks.tryToReplaceScoreboardLine
+import at.hannibal2.skyhanni.test.command.ErrorManager
+import at.hannibal2.skyhanni.utils.CollectionUtils.nextAfter
+import at.hannibal2.skyhanni.utils.LorenzUtils.inAnyIsland
+import at.hannibal2.skyhanni.utils.LorenzUtils.inDungeons
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.NumberUtil.percentageColor
+import at.hannibal2.skyhanni.utils.RenderUtils.HorizontalAlignment
+import at.hannibal2.skyhanni.utils.StringUtils.firstLetterUppercase
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.TabListData
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.TimeUtils.formatted
+import io.github.moulberry.notenoughupdates.util.SkyBlockTime
+import java.util.function.Supplier
+
+private val config get() = SkyHanniMod.feature.gui.customScoreboard
+private val displayConfig get() = config.displayConfig
+private val informationFilteringConfig get() = config.informationFilteringConfig
+
+var unknownLines = listOf<String>()
+var amountOfUnknownLines = 0
```

Comment:
3 things:
* Include these in the companion object (```private``` must be changed to something more open, I used ```internal```)
* Make ```unknownLines``` and ```amountOfUnknownLines``` ```internal``` since they are never used outside of the package
* Make the types explicit, after that you will see a stupid amount of unchecked nulls

the functions in the toplevel can stay since putting them into the companion object only increases indent and does not change anything else since they are all private

Reply:
> Make the types explicit, after that you will see a stupid amount of unchecked nulls
which types, the unknownLines vars?

Reply:
all gets, since Intelji cries that it can't do it's checking with the explicit types.

Reply:
But for now leave it as is. I will fix that later on with a more general solution.